packages/components/src/slot-fill/index.native.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+import Slot from './slot';
+import Fill from './fill';
+import Provider, { Consumer } from './context';
+
+export { Slot };
+export { Fill };
+export { Provider, Consumer };
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BottomSheet,
+} from '@wordpress/components';
+
+export function createSlotFill( name ) {
+	const FillComponent = ( props ) => {
+		return <Fill name={ name } { ...props } />;
+	};
+	FillComponent.displayName = name + 'Fill';
+
+	const SlotComponent = ( props ) => {
+		if ( props.buttons ) {
+			return <Slot name={ name } { ...props } />;
+		}
+		return (
+			<OptionsContainer { ...props.fillProps }>
+				<Slot name={ name } { ...props } />
+			</OptionsContainer>
+		);
+	};
+	SlotComponent.displayName = name + 'Slot';
+
+	const OptionsContainer = ( { children, showSettings, onSettingsToggle: onClose, ...rest } ) => (
+		<BottomSheet
```

Comment:
I didn't understand the need for putting BottomSheet component inside slot-fill. slot-fill is a multi purpose pattern, would it be possible if we don't touch it at all?

Reply:
I agree, I don't see the need to change slot-fill for this. I'd recommend looking at how this works on the web:

- `@wordpress/edit-post` has a `Layout` component that includes a `Sidebar.Slot`
- I believe that the sidebar visibility is controlled via the `core/edit-post` store with the `isEditorSidebarOpened` selector. The button to open the sidebar would dispatch the `openGeneralSidebar` action.
- It also includes a `SettingsSidebar` that fills that slot with many things, including `BlockInspector`.
- This `BlockInspector` is where `InspectorControls.Slot` is placed

For mobile, we use a modal `BottomSheet` instead of the `Sidebar` (although I imagine iPads could show a sidebar in the future), and the button to toggle it is in the block toolbar instead of a global editor toolbar.

The UI is a little different but I think the existing web architecture should still fit the native UI.

Reply:
@koke, @pinarol thank you for review.

I followed with the remarks you have posted and implement changes as below:
1. I have moved `BottomSheet` under `Layout` component (the question here is if we want to wrap it in e.g. <BottomSheet.Slot /> to make it more similar to web or at this stage it will be ok as it is)
2. I use logic from web to show/hide the settings `BottomSheet` container
3. I have created `BlockSettingsButton` and use `slot-fill` logic because I need to conditionally show/hide it depending if given block have some settings wrapped in `InspectorControls`
4 I created native implementation for `InspectorControls` which will render the same as for web plus `SettingsButton`

In current implementation I was able to remove all changes in `slot-fill` logic (only `InspectorControls` for mobile is affected) and also the behaviour is the same as I recorder before. You can noticed that after you delete the children of `InspectorControls` in `edit.native.js` for image block the settings button disappears.